scripts/time_plot.py

```diff
@@ -2181,13 +2181,29 @@ def handle_websock_event(handlerkey,*args):
 #else decodes, eg d0001hd0003v into ('d0001h','d0003v')
 #returns () if otherwise invalid
 #note this is not foolproof
+#ANTNAMEPREFIX='s%04d' #SKA; ANTNAMEPREFIX='m%03d' #meerkat; ANTNAMEPREFIX='ant%d' #kat7
+#decodes 23->m023,m1->m001, s3->s0003
 def decodecustomsignal(signalstr):
     sreg=re.compile('[h|v|H|V|x|y]').split(signalstr)
     if (len(sreg)!=3 or len(sreg[2])!=0):
-        return ();
-    if ((not sreg[0].isdigit()) or (not sreg[1].isdigit())):
-        return (sreg[0]+signalstr[len(sreg[0])],sreg[1]+signalstr[len(sreg[0])+1+len(sreg[1])])
-    return (ANTNAMEPREFIX%(int(sreg[0]))+signalstr[len(sreg[0])].lower(),ANTNAMEPREFIX%(int(sreg[1]))+signalstr[len(sreg[0])+1+len(sreg[1])].lower())
+        return ()
+    if sreg[0].isdigit():
+        rv0='m%03d'%int(sreg[0])
+    elif sreg[0][0].lower()=='m' and sreg[0][1:].isdigit():
```

Comment:
More things that beg for regular expressions...

Reply:
Will skip regex for now. I think other instances where I did use regex is a bit less readable, although I admit this seems a bit clumsy.